### ISSUE TITLE
I'm failing at the HTML5 part of this...

### ISSUE THREAD

**Issue by WNYmathGuy** [bigbluebutton/greenlight#198]
I wish I could say I knew what I was doing, but I kind of don't.

I'm at this phase http://docs.bigbluebutton.org/html/html5-install.html#2-install-html5-client 

I'm new to BBB in general. Last day or so I've been tinkering around with it and almost have the whole thing working fine. I'm on a Ubuntu 16.04 server, and the big difference between my set-up and the instructions at http://docs.bigbluebutton.org/install/install.html , is my server is running a Virtualmin management system. I didn't use a Virtual Server to put BBB on though, and I got through all the tangles of having NGINX get the ports Apache wanted. I'm dying at the HTML5 part though. 

I know that bbb-html5.service isn't running now and doesn't run after reboot even though it's set to. What follows is meaningless to me, so what can I do now?

`:~$ sudo apt-get install -y bbb-html5
Reading package lists... Done
Building dependency tree       
Reading state information... Done
The following NEW packages will be installed:
  bbb-html5
0 upgraded, 1 newly installed, 0 to remove and 0 not upgraded.
Need to get 0 B/31.8 MB of archives.
After this operation, 119 MB of additional disk space will be used.
Selecting previously unselected package bbb-html5.
(Reading database ... 212570 files and directories currently installed.)
Preparing to unpack .../bbb-html5_1%3a1.1.0-184_amd64.deb ...
Unpacking bbb-html5 (1:1.1.0-184) ...
Setting up bbb-html5 (1:1.1.0-184) ...
Installing meteor packages (this may take a moment)
Updating Node packages for bbb-html5 (this may take a moment)
npm WARN read-shrinkwrap This version of npm is compatible with lockfileVersion@1, but npm-shrinkwrap.json was generated for lockfileVersion@0. I'll try to do my best with it!

> meteor-dev-bundle@0.0.0 install /usr/share/meteor/bundle/programs/server
> node npm-rebuild.js

npm WARN lifecycle npm is using /usr/bin/node but there is no node binary in the current PATH. Use the `--scripts-prepend-node-path` option to include the path for the node binary npm was executed with.

> hiredis@0.5.0 install /usr/share/meteor/bundle/programs/server/npm/node_modules/hiredis
> node-gyp rebuild

npm ERR! file sh
npm ERR! path sh
npm ERR! code ELIFECYCLE
npm ERR! errno ENOENT
npm ERR! syscall spawn sh
npm ERR! hiredis@0.5.0 install: `node-gyp rebuild`
npm ERR! spawn sh ENOENT
npm ERR! 
npm ERR! Failed at the hiredis@0.5.0 install script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /usr/share/meteor/.npm/_logs/2018-02-01T06_46_05_926Z-debug.log
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! meteor-dev-bundle@0.0.0 install: `node npm-rebuild.js`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the meteor-dev-bundle@0.0.0 install script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /usr/share/meteor/.npm/_logs/2018-02-01T06_46_05_974Z-debug.log
dpkg: error processing package bbb-html5 (--configure):
 subprocess installed post-installation script returned error exit status 1
Errors were encountered while processing:
 bbb-html5
E: Sub-process /usr/bin/dpkg returned an error code (1)
:~$ `

followed by:
`:~$ sudo apt-get dist-upgrade
Reading package lists... Done
Building dependency tree       
Reading state information... Done
Calculating upgrade... Done
0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
1 not fully installed or removed.
After this operation, 0 B of additional disk space will be used.
Do you want to continue? [Y/n] Y
Setting up bbb-html5 (1:1.1.0-184) ...
Installing meteor packages (this may take a moment)
Updating Node packages for bbb-html5 (this may take a moment)
npm WARN read-shrinkwrap This version of npm is compatible with lockfileVersion@1, but npm-shrinkwrap.json was generated for lockfileVersion@0. I'll try to do my best with it!

> meteor-dev-bundle@0.0.0 install /usr/share/meteor/bundle/programs/server
> node npm-rebuild.js

npm WARN lifecycle npm is using /usr/bin/node but there is no node binary in the current PATH. Use the `--scripts-prepend-node-path` option to include the path for the node binary npm was executed with.

> hiredis@0.5.0 install /usr/share/meteor/bundle/programs/server/npm/node_modules/hiredis
> node-gyp rebuild

npm ERR! file sh
npm ERR! path sh
npm ERR! code ELIFECYCLE
npm ERR! errno ENOENT
npm ERR! syscall spawn sh
npm ERR! hiredis@0.5.0 install: `node-gyp rebuild`
npm ERR! spawn sh ENOENT
npm ERR! 
npm ERR! Failed at the hiredis@0.5.0 install script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /usr/share/meteor/.npm/_logs/2018-02-01T06_54_28_776Z-debug.log
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! meteor-dev-bundle@0.0.0 install: `node npm-rebuild.js`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the meteor-dev-bundle@0.0.0 install script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /usr/share/meteor/.npm/_logs/2018-02-01T06_54_28_806Z-debug.log
dpkg: error processing package bbb-html5 (--configure):
 subprocess installed post-installation script returned error exit status 1
Errors were encountered while processing:
 bbb-html5
E: Sub-process /usr/bin/dpkg returned an error code (1)
:~$ `

**Comment by joshua-arts** [comment]
Hey @WNYmathGuy,

This is the repository for GreenLight (a simple front-end for your BigBlueButton server). Your question seems to be more about running BigBlueButton itself.

You're probably better off asking your question in [bbb-setup Google group](https://groups.google.com/forum/#!forum/bigbluebutton-setup). You should be able to get much more help there!

Thanks!

**Comment by WNYmathGuy** [comment]
It is about Greenlight because this problem arose while following the greenlight install instructions. It's the greenlight software that permits the mobile use of BBB via the HTML5 thing. Ubuntu 16.04 seems to have a problem with all of that right now. Not sure if it's just ripple effect of security problems lately, but it's not working and my only connection to HTML5 is the Greenlight software.

**Comment by WNYmathGuy** [comment]
Also, the Google OAuth2 fails. I followed the instructions best I could. Twitter works fine, but Google fails. I get past the point where Google gives a window to select which Google account you want to go in as, but then theirs a failure message after.
https://bbb.ruppssites.com/AccessDenied.png
...and also this doesn't seem to work
LANDING_BACKGROUND=https://bbb.ruppssites.com/UbuntuBlack.jpg

**Comment by joshua-arts** [comment]
You said you're following [these](http://docs.bigbluebutton.org/html/html5-install.html#2-install-html5-client) steps? Those are the steps to install the HTML5 Client. You should definitely ask this question in the Google group.

In terms of the Google OAuth2, are you certain you've setup your credentials correctly? Also, if you're setting `GOOGLE_OAUTH_HD` then emails that don't match that domain will raise invalid credentials.

**Comment by WNYmathGuy** [comment]
I applied to that group before posting here, but they haven't let me in yet.

My `"GOOGLE_OAUTH2_HD="` variable is left blank. I followed the advice and set-up a new gmail account just for mailing from this server. The credentials are `"GOOGLE_OAUTH2_ID=_<long random looking string>_.apps.googleusercontent.com"` and `"GOOGLE_OAUTH2_SECRET=_<definitely a random string>_"`. All the SMTP mailer settings are working with that email address.
